cwr/index_test.go

```diff
@@ -48,194 +50,192 @@ func (t *testIndex) cleanup() {
+		iswc               string
+		publisherSequenceN string
+	)
+
+	row, err := x.db.Query(`SELECT object_id,title,iswc FROM registered_work WHERE cwr_id = ?`, x.cwrCid.String())
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer x.cleanup()
-	// check all the publisherControlledBySubmitter were indexed
-	for _, record := range x.records {
-		// check the publisher_sequence_number indexes
-		if record.RecordType != "SPU" {
-			continue
+	for row.Next() {
+
+		if err = row.Scan(&objectID, &title, &iswc); err != nil {
+			t.Fatal(err)
+		}
+		if err = check(objectID, map[string]string{
+			"title": title,
+			"iswc":  iswc,
+		}); err != nil {
+			t.Fatal(err)
 		}
-		rows, err := x.db.Query(
-			`SELECT object_id FROM publisher_control WHERE publisher_sequence_n = ? AND record_type = ?`,
-			record.PublisherSequenceNumber, record.RecordType,
-		)
+		//get all publisher control records (SPU) which link to the above tx .
+		txID, err := cid.Decode(objectID)
+		if err != nil {
+			t.Fatal(err)
+		}
+		rows, err := x.db.Query(`SELECT object_id,publisher_sequence_n FROM publisher_control WHERE tx_id = ?`, txID.String())
 		if err != nil {
 			t.Fatal(err)
 		}
-		defer rows.Close()
-		var objectID string
 		for rows.Next() {
-			// if we've already set objectID then we have a duplicate
-			if objectID != "" {
-				t.Fatalf("duplicate entries for SPU publisher sequence number %q", record.PublisherSequenceNumber)
-			}
-			if err := rows.Scan(&objectID); err != nil {
-				t.Fatal(err)
-			}
 
-			// check we can get the object from the store
-			cid, err := cid.Parse(objectID)
-			if err != nil {
+			if err := rows.Scan(&objectID, &publisherSequenceN); err != nil {
 				t.Fatal(err)
 			}
-			obj, err := x.store.Get(cid)
-			if err != nil {
+			if err = check(objectID, map[string]string{
+				"publisher_sequence_n": publisherSequenceN,
+			}); err != nil {
 				t.Fatal(err)
 			}
-			// check the object has the correct fields
-			for key, expected := range map[string]string{
-				"publisher_sequence_n": record.PublisherSequenceNumber,
-				"record_type":          record.RecordType,
-			} {
-				actual, err := obj.GetString(key)
-				if err != nil {
-					t.Fatal(err)
-				}
-				if actual != expected {
-					t.Fatalf("expected object %s to be %q, got %q", key, expected, actual)
-				}
-			}
 		}
+	}
+}
+func openStore() (*meta.Store, error) {
+	metaDir := ".meta"
+	if err := os.MkdirAll(metaDir, 0755); err != nil {
+		return nil, err
+	}
+	store, err := fs.NewDatastore(metaDir)
+	if err != nil {
+		return nil, err
+	}
+	return meta.NewStore(store), nil
+}
 
-		// check we got an object and no db errors
-		if objectID == "" {
-			t.Fatalf("SPU %q not found", record.PublisherSequenceNumber)
-		} else if err := rows.Err(); err != nil {
-			t.Fatal(err)
+func newTestIndex() (x *testIndex, err error) {
+	// convert the test cwr to META object
+	x = &testIndex{}
+	defer func() {
+		if err != nil {
+			x.cleanup()
 		}
+	}()
+
+	x.store, err = openStore()
```

Comment:
Why are we using a filesystem store in the tests? If there is a valid reason, it should be created in a temporary directory which is deleted on exit.

Reply:
We use filesystem store in the test because the in memory store map does not support concurrency.
Will create a temp dir and delete on exit.

Reply:
Ok, we could just implement a thread-safe map store?

Reply:
Sure we can.
Does it worth the effort ? currently it is based on ipfs ds.
Consider that soon the test will include swarm datastore (test simulation)…

Reply:
Ok let's just use the file store :+1: